### ISSUE TITLE
Batchnorm does not work in Eager mode in TF 1.12: InternalError

### ISSUE THREAD

**Issue by ameyinvent** [tensorflow/tensorflow#25814]
Could not find valid device for node

**Comment by jvishnuvardhan** [comment]
@ameyinvent Please provide the details requested in the [template](https://github.com/tensorflow/tensorflow/issues/new?template=00-bug-performance-issue.md). Please provide as many details as possible to find rootcause of the bug. Also, check this [resource](https://github.com/tensorflow/tensorflow/issues/25525) that might help you. Thanks!

**Comment by jvishnuvardhan** [comment]
@ameyinvent Could you provide a code to reproduce the bug? If this was resolved already, please close the issue or let me know. Thanks!

**Comment by jvishnuvardhan** [comment]
I think it was resolved. Closing due to lack of recent activity. Please open new ticket if you see similar issue. Thanks!
